### ISSUE TITLE
We are unable to filter using binary hex value

### ISSUE THREAD

**Issue by adkhare** [youtube/vitess#2235]
```select * from table where id = 0x42398538hiefif834``` does not work.

It would be great to have it work so that we can use it query from some of the UI SQL tools.

**Comment by sougou** [comment]
The documentation for hex literals is here: http://dev.mysql.com/doc/refman/5.7/en/hexadecimal-literals.html.

However, it's not the complete story. In reality, the interpretation of a `0x` notation is context dependent. Here is an illustration:
```
MariaDB [a]> create table a(id int, val varchar(128), primary key(id));
Query OK, 0 rows affected (0.03 sec)

MariaDB [a]> insert into a values(0x31, 0x31);
Query OK, 1 row affected (0.00 sec)

MariaDB [a]> select * from a;
+----+------+
| id | val  |
+----+------+
| 49 | 1    |
+----+------+
1 row in set (0.00 sec)

MariaDB [a]> insert into a values(49, 49);
Query OK, 1 row affected (0.01 sec)

MariaDB [a]> select * from a;
+----+------+
| id | val  |
+----+------+
| 49 | 49   |
+----+------+
1 row in set (0.01 sec)

MariaDB [a]> insert into a values('1', '1');
Query OK, 1 row affected (0.00 sec)

MariaDB [a]> select * from a;
+----+------+
| id | val  |
+----+------+
|  1 | 1    |
+----+------+
1 row in set (0.00 sec)

```

In the above example, `0x31` is the decimal number `49`, which is the ascii representation of `'1'`. If `0x31` is converted as `'1'` as specified by the spec, a number column interprets it as `1` instead of `49`. On the other hand, if it's converted to the number `49`, the varchar column treats it as `'49'`.
In other words, there's no way to interpret a `0x` literal correctly without knowing the target data type.

Fortunately, the `X'...'` notation seems to have a target data type agnostic interpretation:

```
MariaDB [a]> insert into a values(X'31', X'31');
Query OK, 1 row affected (0.00 sec)

MariaDB [a]> select * from a;
+----+------+
| id | val  |
+----+------+
|  1 | 1    |
+----+------+
1 row in set (0.00 sec)

```

Long story short, it should be trivial to support the `X'...'` notation. But I'll stay away from the `0x` notation because it has too many pitfalls.

**Comment by adkhare** [comment]
@sougou Thanks for looking into this. we are perfectly good with support for `x'...' ` notation.

**Comment by sougou** [comment]
It turns out that the parser was allowing `0x` constructs, as well as treating `0...` as octal, which is not MySQL behavior. I'll remove support for `0x` and treat `0...` as decimal just like MySQL does.
